src/python/pants/pantsd/service/scheduler_service.py

```diff
@@ -65,20 +67,32 @@ def __init__(
   def _combined_invalidating_fileset_from_globs(glob_strs, root):
     return set.union(*(Fileset.globs(glob_str, root=root)() for glob_str in glob_strs))
 
+  def _get_snapshot(self, glob):
+    """Returns a Snapshot of the input glob"""
+    return self._scheduler.capture_snapshots(
+      (PathGlobsAndRoot(PathGlobs(glob), self._build_root), ))[0]
+
+  def _get_files(self, snapshots):
+    """Returns a list of the files corresponding to the list of input snapshots"""
+    return [snapshot.files[0] for snapshot in snapshots]
+
   def setup(self, services):
     """Service setup."""
     super(SchedulerService, self).setup(services)
     # Register filesystem event handlers on an FSEventService instance.
     self._fs_event_service.register_all_files_handler(self._enqueue_fs_event)
 
-    # N.B. We compute this combined set eagerly at launch with an assumption that files
+    # N.B. We compute the invalidating fileset eagerly at launch with an assumption that files
     # that exist at startup are the only ones that can affect the running daemon.
     if self._invalidation_globs:
-      self._invalidating_files = self._combined_invalidating_fileset_from_globs(
+      invalidating_files = self._combined_invalidating_fileset_from_globs(
         self._invalidation_globs,
         self._build_root
       )
-    self._logger.info('watching invalidating files: {}'.format(self._invalidating_files))
+      for glob in invalidating_files:
+        self._snapshot_by_glob[glob] = self._get_snapshot([glob])
```

Comment:
`PathGlobs` takes multiple globs values, so rather than a loop here, you'll want to load all the globs at once: that will mean just one snapshot, which will be more efficient.